### PR TITLE
Alerting: Remove unused code after importing from grafana/alerting

### PR DESCRIPTION
**Pull request by gotjosh**
Negative diff with integration tests passing - just what I like to see.

### CODE REVIEW

File: pkg/services/ngalert/notifier/alertmanager.go [hunk 1]
```diff
@@ -8,29 +7,12 @@ import (
...
-	"github.com/prometheus/alertmanager/config"
-	"github.com/prometheus/alertmanager/dispatch"
-	"github.com/prometheus/alertmanager/inhibit"
-	"github.com/prometheus/alertmanager/nflog"
-	"github.com/prometheus/alertmanager/nflog/nflogpb"
-	"github.com/prometheus/alertmanager/notify"
-	"github.com/prometheus/alertmanager/provider/mem"
-	"github.com/prometheus/alertmanager/silence"
-	"github.com/prometheus/alertmanager/timeinterval"
-	"github.com/prometheus/alertmanager/types"
```
Comment (gotjosh) [inline pkg/services/ngalert/notifier/alertmanager.go:28]: 🙏

File: pkg/services/ngalert/notifier/alertmanager_test.go [hunk 2]
```diff
@@ -53,367 +45,7 @@ func setupAMTest(t *testing.T) *Alertmanager {
...
-		return len(found) == 3
-	}, 3*time.Second, 100*time.Millisecond)
-
-	// Wait again for another silence to expire.
-	require.Eventually(func() bool {
-		found, err := am.ListSilences(nil)
-		require.NoError(err)
-		return len(found) == 2
-	}, 6*time.Second, 100*time.Millisecond)
+	require.False(t, am.Ready())
```
Comment (grobinson-grafana) [inline pkg/services/ngalert/notifier/alertmanager_test.go:50]: What is the purpose of this test, to assert that the Alertmanager is not ready after `setupAMTest`?

**Review by grobinson-grafana — APPROVED** [review]

File: pkg/services/ngalert/notifier/alertmanager_test.go (hunk 2 above)
Comment (gotjosh) [reply pkg/services/ngalert/notifier/alertmanager_test.go:50]: Yeah, it assets that the construction of the struct was a "success". A newly created AM shouldn't be ready until a configuration is applied to it.

File: pkg/services/ngalert/notifier/alertmanager.go [hunk 3]
```diff
@@ -203,16 +150,12 @@ func (am *Alertmanager) Ready() bool {
...
-	return am.config != nil
-}
-
 func (am *Alertmanager) StopAndWait() {
 	am.Base.StopAndWait()
 }
 
 // SaveAndApplyDefaultConfig saves the default configuration the database and applies the configuration to the Alertmanager.
-// It rollbacks the save if we fail to apply the configuration.
+// It rollsback the save if we fail to apply the configuration.
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/alertmanager.go:158]:
```suggestion
// It rolls back the save if we fail to apply the configuration.
```
nitpickiest of nits

**Review by alexweav — APPROVED** [review]

**Pull request merged**